task_templates/pipelines/python3_sparse/custom.py

```diff
@@ -15,133 +19,91 @@ def fit(
     row_weights: Optional[np.ndarray] = None,
     **kwargs,
 ):
-    """
-    This hook must be implemented with your fitting code, for running drum in the fit mode.
+    """ This hook MUST ALWAYS be implemented for custom tasks.
+
+    This hook defines how DataRobot will train this task.
+    DataRobot runs this hook when the task is being trained inside a blueprint.
 
-    This hook MUST ALWAYS be implemented for custom tasks.
-    For inference models, this hook can stick around unimplemented, and won’t be triggered.
+    DataRobot will pass the training data, project target, and additional parameters based on the project
+    and blueprint configuration as parameters to this function.
+
+    As an output, this hook is expected to create an artifact containing a trained object,
+    that is then used to score new data.
 
     Parameters
     ----------
-    X: pd.DataFrame - training data to perform fit on
-    y: pd.Series - target data to perform fit on
-    output_dir: the path to write output. This is the path provided in '--output' parameter of the
-        'drum fit' command.
-    class_order : A two element long list dictating the order of classes which should be used for
-        modeling. Class order will always be passed to fit by DataRobot for classification tasks,
+    X: pd.DataFrame
+        Training data that DataRobot passes when this task is being trained. Note that both the training data AND
+        column (feature) names are passed
```

Comment:
should mention this is sparse and that it's COO format

Reply:
good call, done